### ISSUE TITLE
[Code health] Replace Date with long throughout

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#292]
`Date` is mutable and isn't thread-safe. Since we only use `Date` for timestamps, these can be replaced with simple longs.

**Comment by nikhil6624** [comment]
@gino-m , may I work on this issue?

**Comment by gino-m** [comment]
> @gino-m , may I work on this issue?

Absolutely! Please be sure to [accept our CLA](https://cla.developers.google.com/) first.

**Comment by nikhil6624** [comment]
@gino-m , I already did it when I was working on another project under Google repo (Google Photos).

**Comment by scolsen** [comment]
@scolsen to follow up.
